econml/validate/utils.py

```diff
@@ -81,14 +85,25 @@ def calc_qini_coeff(
     for it in range(len(qs)):
         inds = (qs[it] <= cate_preds_val)  # group with larger CATE prediction than the q-th quantile
         group_prob = np.sum(inds) / n  # fraction of population in this group
-        toc[it] = group_prob * (
-            np.mean(dr_val[inds]) - ate)  # tau(q) = q * E[Y(1) - Y(0) | tau(X) >= q[it]] - E[Y(1) - Y(0)]
-        toc_psi[it, :] = np.squeeze(
-            (dr_val - ate) * (inds - group_prob) - toc[it])  # influence function for the tau(q)
+        if metric == 'qini':
+            toc[it] = group_prob * (
+                np.mean(dr_val[inds]) - ate)  # tau(q) = q * E[Y(1) - Y(0) | tau(X) >= q[it]] - E[Y(1) - Y(0)]
+            toc_psi[it, :] = np.squeeze(
+                (dr_val - ate) * (inds - group_prob) - toc[it])  # influence function for the tau(q)
+        else:
```

Comment:
It's worth asserting that the metric is `"toc"` as expected in case the user incorrectly passed something else.

Reply:
Done, addressed in latest commits.